diff -Nur old/configure.in new/configure.in
--- old/configure.in	2004-12-09 01:25:41.000000000 +0000
+++ new/configure.in	2005-04-14 00:09:45.000000000 +0000
@@ -53,7 +53,7 @@
 fi
 
 $COURIERAUTHCONFIG --configfiles | sed 's/^/auth/' >configtmp
-. configtmp
+. `pwd`/configtmp
 courierlogger=$authsbindir/courierlogger
 rm -f configtmp
 AC_SUBST(courierlogger)
diff -Nur old/Makefile.am new/Makefile.am
--- old/Makefile.am	2004-12-09 01:25:41.000000000 +0000
+++ new/Makefile.am	2005-04-14 00:14:41.000000000 +0000
@@ -116,7 +116,7 @@
 
 socksify: socksify.in libsockswrap.la
 	CONFIG_FILES=$@ CONFIG_HEADERS= $(SHELL) ./config.status
-	. libsockswrap.la || exit 1 ; sed 's/@LIBNAME@/'$${dlname}'/' <socksify >socksify.tmp
+	. `pwd`/libsockswrap.la || exit 1 ; sed 's/@LIBNAME@/'$${dlname}'/' <socksify >socksify.tmp
 	mv socksify.tmp socksify
 
 check-am:
